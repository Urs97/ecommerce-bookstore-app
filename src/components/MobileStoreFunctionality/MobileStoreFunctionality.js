import React, { useContext } from 'react';
import './MobileStoreFunctionality.scss';

import StoreContext from '../../context/StoreContext';
import MobileFilter from '../MobileFilter/MobileFilter';

const MobileStoreFunctionality = () => {
    const context = useContext(StoreContext);
    const total_books = context.currentData;

    return (
        <section className="mobile-store-functionality">
            <span>{total_books.length} products</span>
            <button 
                className="button mobile-filter-open" 
                onClick={() => context.toggleMobileFilter('open')}
            >Filter</button>
            <MobileFilter />
        </section>
    )
};

export default MobileStoreFunctionality;