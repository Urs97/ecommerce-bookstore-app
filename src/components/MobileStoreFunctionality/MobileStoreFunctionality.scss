@import "../../styles/shared.scss";

.mobile-store-functionality {
  margin-bottom: 3em;
  border-radius: $border-radius;
  display: flex;
  align-items: center;
  justify-content: center;
  display: none;

  & .mobile-filter-open {
    font-size: 1.25em;
    background-color: $dark-blue;
    margin-left: 2em;
  }

  & .mobile-.mobile-filter-open {
    background-color: $yellow;
  }

  span {
    font-size: 1.75em;
  }
}

@media (max-width: 800px) {
  .mobile-store-functionality {
    display: flex;
  }
}

@media (max-width: 450px) {
  .mobile-store-functionality {
    font-size: 1.25em;
  }
}
