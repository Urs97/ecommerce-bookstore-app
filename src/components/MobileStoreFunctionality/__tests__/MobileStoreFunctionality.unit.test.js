import React from 'react';
import {shallow} from 'enzyme';
import MobileStoreFunctionality from '../MobileStoreFunctionality';

describe('MobileStoreFunctionality', () => {
    test('renders MobileStoreFunctionality', () => {
        const wrapper = shallow(
            <MobileStoreFunctionality />
        );
        expect(wrapper).toMatchSnapshot();
    });
});