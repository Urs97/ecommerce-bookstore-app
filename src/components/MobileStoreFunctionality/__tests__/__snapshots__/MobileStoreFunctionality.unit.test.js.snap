// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileStoreFunctionality renders MobileStoreFunctionality 1`] = `
<section
  className="mobile-store-functionality"
>
  <span>
    0
     products
  </span>
  <button
    className="button mobile-filter-open"
    onClick={[Function]}
  >
    Filter
  </button>
  <MobileFilter />
</section>
`;
